projects/rocprofiler-systems/source/lib/rocprof-sys/library/components/category_region.hpp

```diff
@@ -28,17 +28,89 @@
 #include "core/timemory.hpp"
 #include "library/causal/data.hpp"
 #include "library/runtime.hpp"
+#include "library/thread_info.hpp"
 #include "library/tracing.hpp"
 #include "library/tracing/annotation.hpp"
 
+#include <map>
+#include <thread>
 #include <timemory/components/gotcha/backends.hpp>
 #include <timemory/hash/types.hpp>
 #include <timemory/mpl/concepts.hpp>
 #include <timemory/mpl/types.hpp>
 #include <timemory/utility/types.hpp>
 
+#include "core/trace_cache/cache_manager.hpp"
+
 #include <string_view>
 
+#include <utility>
+
+namespace
+{
+
+void
+cache_region(uint64_t thread_id, const std::string& name, uint64_t start_ts,
+             uint64_t end_ts, const std::string& category)
+{
+    size_t zero = 0;
+    rocprofsys::trace_cache::get_buffer_storage().store(
+        rocprofsys::trace_cache::entry_type::region, thread_id, name.c_str(), zero, zero,
+        start_ts, end_ts, "", "", category.c_str());
+}
+
+struct entry_key
+{
+    std::string name;
+    std::string category;
+
+    friend bool operator<(const entry_key& lhs, const entry_key& rhs)
+    {
+        return lhs.name < rhs.name && lhs.category < rhs.category;
+    }
+};
+
+using timestamp_t = long;
```

Comment:
```suggestion
using timestamp_t = uint64_t;
```
`uint64_t` would be a better choice. This is what `cache_region` is expecting.